pennylane/qchem/molecule.py

```diff
@@ -121,7 +121,7 @@ def __init__(
 
         if unit == "Angstrom":
             self.coordinates = self.coordinates / bohr_angs
-        elif unit != "Angstrom" and unit != "Bohr":
+        elif unit not in ("Angstrom", "Bohr"):
```

Comment:
We should always aim to fail early, i.e., we should shift this right under the checks done above for atoms and basis set.